### ISSUE TITLE
Dokku scripts doesn't work with dockerfile deployment

### ISSUE THREAD

**Issue by abellion** [dokku/dokku#2300]
Hi,

I'm trying to run a simple command during the predeploy phase with an app.json at the root of my repository but nothing happen, I always get the following : 

![image](https://cloud.githubusercontent.com/assets/9916488/16624008/614fc3ce-43a0-11e6-9e1b-2cb00e269535.png)

I have put an app.json at the root of my repo, an other one inside an app/ folder, and a scripts.dokku at the root, but no one work.

```
{
  "name": "jubiwee-api",
  "description": "Jubiwee core API",
  "scripts": {
    "dokku": {
      "predeploy": "touch /app/predeploy.test",
    }
  },
  "env": {
    "APP_ENV": "production",
    "APP_DEBUG": "false",
    "CACHE_DRIVER": "memcached",
    "QUEUE_DRIVER": "sync"
  }
}
```

- `uname -a`: ```Linux scw-d78731 4.5.1-std-1 #1 SMP Mon Apr 18 19:58:58 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux```
- `free -m`: 
![image](https://cloud.githubusercontent.com/assets/9916488/16625068/636284be-43a3-11e6-841d-1ec1bb2622ca.png)

- `docker version`: 
![image](https://cloud.githubusercontent.com/assets/9916488/16625083/74c15410-43a3-11e6-9204-175175103dbb.png)

- `docker -D info`: 
![image](https://cloud.githubusercontent.com/assets/9916488/16625128/a2573796-43a3-11e6-95e2-8a4f723563e4.png)

- `sigil -v`: ```0.4.0```
- `docker run -ti gliderlabs/herokuish:latest herokuish version`: 
![image](https://cloud.githubusercontent.com/assets/9916488/16625222/c00e7cfe-43a3-11e6-8e6c-3e5cfee72f02.png)

- `dokku version`: ```0.6.4```
- `dokku plugin`: 
![image](https://cloud.githubusercontent.com/assets/9916488/16625242/d8b6f68c-43a3-11e6-857a-13f5b7ff4d5b.png)

Dokku was installed through the standard way described in the doc.

Thanks

**Comment by josegonzalez** [comment]
Can you include *all* of the information we ask for in our [issue template](https://github.com/dokku/dokku/blob/master/ISSUE_TEMPLATE.md)? It is helpful for debugging problems. Thanks.

**Comment by abellion** [comment]
Hi,

Yeah sorry, I've missed that ! My first message is edited with the requested infos !

**Comment by michaelshobbs** [comment]
Can you also share your dockerfile?

**Comment by abellion** [comment]
Of course : 

```
FROM php:7.0-apache
MAINTAINER Antoine Bellion <---@domain.com>

# Update system and install libs
RUN apt-get update \
&& apt-get install -y libssl-dev git \
&& apt-get clean

# Install php mongodb ext
RUN pecl install mongodb && docker-php-ext-enable mongodb

# Install php mbstring and zip ext
RUN docker-php-ext-install mbstring zip

# Enable apache rewrite module
RUN a2enmod rewrite

# Copy project files from host to container
COPY . /var/www/html/

# Grant rights on the projet directory
WORKDIR /var/www/html
RUN usermod -u 1000 www-data \
&& chown -R www-data /var/www/html \
&& chown -R www-data /var/www/html/storage

# Install composer and project dependencies
RUN curl -sS https://getcomposer.org/installer | php
RUN mv composer.phar /usr/local/bin/composer

EXPOSE 80
```

**Comment by josegonzalez** [comment]
Can you share the output of a deploy after running `dokku trace on`?

**Comment by abellion** [comment]
Yes, here it is : http://pastebin.com/Bu0HCL2g

And this is my application structure : 
![image](https://cloud.githubusercontent.com/assets/9916488/16626392/f1cd5634-43a8-11e6-87fd-5525686c4f19.png)

As you can see there is a "app.json" and "scripts.dokku" both in the root and in the "app" folder.

**Comment by michaelshobbs** [comment]
In dockerfile apps we look for `app.json` in the defined `WORKDIR` first then `/apps` if `WORKDIR` is not defined. @josegonzalez [this section](http://dokku.viewdocs.io/dokku/advanced-usage/deployment-tasks/#appjson-and-scriptsdokku) probably could use an update to reflect that.

Unfortunately we seem to have hit a bash limitation (perhaps) where we're not seeing the trace output of the sub-subshell execution of [`get_json_value`](https://github.com/dokku/dokku/blob/v0.6.4/plugins/00_dokku-standard/exec-app-json-scripts#L16). I verified this on a dockerfile app that does seem to work.

Applying the following debug patch will at least show us if we're finding the `app.json file`
```
diff --git a/plugins/00_dokku-standard/exec-app-json-scripts b/plugins/00_dokku-standard/exec-app-json-scripts
index 3f7700f..cb0fa49 100755
--- a/plugins/00_dokku-standard/exec-app-json-scripts
+++ b/plugins/00_dokku-standard/exec-app-json-scripts
@@ -13,6 +13,7 @@ get_phase_script() {
   copy_from_image "$IMAGE" "/app/app.json" "$GET_PHASE_SCRIPT_TMP_WORK_DIR" 2>/dev/null || true

   if [[ -f "$APP_JSON_FILE" ]];then
+    cat "$APP_JSON_FILE" 2>/dev/null
     local VALUE=$(get_json_value "scripts.dokku.${PHASE_SCRIPT_KEY}" < "$APP_JSON_FILE")
   else
     return 0
```

**Comment by michaelshobbs** [comment]
Ah! I see the issue and was able to reproduce it. Simple fix forthcoming.

**Comment by josegonzalez** [comment]
@michaelhobbs mind updating the docs while you're at it?

**Comment by abellion** [comment]
Good to know, thank you for your fast support !
Is there any workaround at the moment ? I just need to execute a command during the pre-deploy phase !

**Comment by josegonzalez** [comment]
@abellion I'll probably make a release today once this hits master.

**Comment by michaelshobbs** [comment]
This patch is the fix

```
diff --git a/plugins/00_dokku-standard/exec-app-json-scripts b/plugins/00_dokku-standard/exec-app-json-scripts
index ab2207a..b6a6870 100755
--- a/plugins/00_dokku-standard/exec-app-json-scripts
+++ b/plugins/00_dokku-standard/exec-app-json-scripts
@@ -10,7 +10,7 @@ get_phase_script() {
   local APP_JSON_FILE="$GET_PHASE_SCRIPT_TMP_WORK_DIR/app.json"
   trap 'rm -rf "$GET_PHASE_SCRIPT_TMP_WORK_DIR" > /dev/null' RETURN INT TERM

-  copy_from_image "$IMAGE" "/app/app.json" "$GET_PHASE_SCRIPT_TMP_WORK_DIR" 2>/dev/null || true
+  copy_from_image "$IMAGE" "app.json" "$GET_PHASE_SCRIPT_TMP_WORK_DIR" 2>/dev/null || true

   if [[ -f "$APP_JSON_FILE" ]];then
     local VALUE=$(get_json_value "scripts.dokku.${PHASE_SCRIPT_KEY}" < "$APP_JSON_FILE")
```

**Comment by abellion** [comment]
Works like a charm ! Thanks guys for your work !

**Comment by josegonzalez** [comment]
@abellion mind filling out this issue for us? https://github.com/dokku/dokku/issues/1878

**Comment by abellion** [comment]
Yes with pleasure !

**Comment by josegonzalez** [comment]
@abellion Released in 0.6.5.
